### PR TITLE
Integrate apk-mitm natively

### PR DESCRIPTION
**Pull request by shroudedcode**
My changes as (mostly) discussed on Telegram. :sparkles:

### CODE REVIEW

File: src/mitm-patches.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            `Using apk-mitm v${APK_MITM_VERSION} (https://github.com/shroudedcode/apk-mitm)\n`
+        );
+
+        const decodeDir = path.dirname(apktoolYmlPath);
+
+        await apkMitm
+            .observeListr(apkMitm.applyPatches(decodeDir))
+            .forEach((line) => outputChannel.appendLine(line));
+
+        quickPickUtil.setQuickPickDefault("rebuildQuickPickItems", "--debug");
```
Comment (Surendrajat) [inline src/mitm-patches.ts:33]: I think this can be removed now.

File: package.json [hunk 2]
```diff
@@ -197,6 +198,7 @@
         "webpack-cli": "^4.2.0"
     },
     "dependencies": {
+        "apk-mitm": "^0.10.1",
         "escape-string-regexp": "^4.0.0",
```
Comment (Surendrajat) [inline package.json:202]: This dep can be removed now.

File: package.json [hunk 3]
```diff
@@ -197,6 +198,7 @@
...
         "extract-zip": "^2.0.1",
         "globby": "^11.0.1",
```
Comment (Surendrajat) [inline package.json:204]: these deps too...

File: webpack.config.js [hunk 4]
```diff
@@ -35,5 +36,17 @@ const config = {
             },
         ],
     },
+    plugins: [
+        new webpack.NormalModuleReplacementPlugin(
+            /^any-observable$/,
+            // See the file for why this is necessary
+            path.join(__dirname, "src/any-observable-fix.js")
```
Comment (Surendrajat) [inline webpack.config.js:43]: Still an ugly hack. Hoping to see it removed someday :)

File: src/any-observable-fix.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// This file is necessary because webpack will fail to load the `any-observable`
+// package (which is a transitive dependency of `apk-mitm`) otherwise. This file
+// replaces that module and hard-codes the `Observable` implementation from
+// `rxjs` as the one that is used by packages which import `any-observable`.
+
+// eslint-disable-next-line import/no-default-export
+export { Observable as default } from "rxjs";
```
Comment (Surendrajat) [inline src/any-observable-fix.js:7]: Does it have to be a `.js` file? Can we make it a `.ts`?

**Review by Surendrajat — CHANGES_REQUESTED** [review]
Thanks for this PR @shroudedcode!
I've left some comments here and there. :)

File: src/mitm-patches.ts (hunk 1 above)
Comment (shroudedcode) [reply src/mitm-patches.ts:33]: True, will do! :+1:

File: src/any-observable-fix.js (hunk 5 above)
Comment (shroudedcode) [reply src/any-observable-fix.js:7]: Not sure, I'll check.

**Review by Surendrajat — APPROVED** [review]
Looks good.
However, I noticed that you're printing relative paths of smali files in output. It will be nice to see the full path printed so that VS Code can show it as a clickable link and the user can open the file directly.

**Review by amsharma44 — APPROVED** [review]
Looks good, thanks @shroudedcode.

**Comment by Surendrajat** [comment]
Thanks, @shroudedcode for your contribution and apk-mitm. Hope to see nice improvements in it :)

**Pull request merged**